holoviews/element/geom.py

```diff
@@ -52,6 +52,37 @@ class VectorField(Selection2DExpr, Geometry):
     vdims = param.List(default=[Dimension('Angle', cyclic=True, range=(0,2*np.pi)),
                                 Dimension('Magnitude')], bounds=(1, None))
 
+    @classmethod
+    def from_uv(cls, data, kdims=None, vdims=None, **params):
```

Comment:
`data` here could be one of N many formats and handling them all is basically impossible. What I would suggest is that you create a Dataset from the kdims and vdims and then use `.dimension_values()` on it to look up the arrays.